src/PaperG/FirehoundBlob/Facebook/Targeting/FacebookGeographicTargeting.php

```diff
@@ -199,6 +199,21 @@ public function getRegionIds()
         return $this->regionIds;
     }
 
+    public function isValid()
+    {
+        if (!(
+            (!empty($this->getCountryAction()) && !empty($this->getCountryIds())) ||
+            (!empty($this->getCityAction()) && !empty($this->getCityIds())) ||
+            (!empty($this->getRegionAction()) && !empty($this->getRegionIds())) ||
+            (!empty($this->getPostalAction()) && !empty($this->getPostalCodes()))
+        )
+        ) {
+            return false;
```

Comment:
instead of the inversion, then returning false, couldn't you just:

`return (
            (!empty($this->getCountryAction()) && !empty($this->getCountryIds())) ||
            (!empty($this->getCityAction()) && !empty($this->getCityIds())) ||
            (!empty($this->getRegionAction()) && !empty($this->getRegionIds())) ||
            (!empty($this->getPostalAction()) && !empty($this->getPostalCodes()))
        ) ;` ?